tests/serialization/json/Converters/JsonDateConverter.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Kiota.Abstractions;
+
+namespace Microsoft.Kiota.Serialization.Json.Tests.Converters;
+
+/// <summary>
+/// Converts a Date object or value to/from JSON.
+/// </summary>
+public class JsonDateConverter : JsonConverter<Date>
+{
+    /// <inheritdoc />
+    public override Date Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        => reader.TokenType == JsonTokenType.Null
```

Comment:
Isn't this reversed?

Reply:
Seems to me correct, TokenType is string and in that case ReadInternal is called